src/bun.js/api/js_zlib.zig

```diff
@@ -155,6 +155,70 @@ pub const ZlibEncoder = struct {
         return this.collectOutputValue();
     }
 
+    pub fn transformWith(this: *@This(), globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) callconv(.C) JSC.JSValue {
+        const arguments = callframe.arguments(4);
+        const vm = globalThis.bunVM();
+        const event_loop = vm.eventLoop();
+
+        if (arguments.len < 4) {
+            globalThis.throwNotEnoughArguments("ZlibEncoder.encode", 4, arguments.len);
+            return .zero;
+        }
+
+        if (this.has_called_end) {
+            globalThis.throw("ZlibEncoder.encodeSync called after ZlibEncoder.end", .{});
+            return .zero;
+        }
+
+        const input = callframe.argument(0);
+        const optional_encoding = callframe.argument(1);
+        const thisctx = arguments.ptr[2];
+        const is_last = callframe.argument(3).toBoolean();
+
+        const push_fn = thisctx.get(globalThis, "push") orelse {
+            globalThis.throw("are you sure this is a stream.Transform?", .{});
+            return .zero;
+        };
+        if (globalThis.hasException()) return .zero;
+
+        const input_to_queue = JSC.Node.BlobOrStringOrBuffer.fromJSWithEncodingValueMaybeAsync(globalThis, bun.default_allocator, input, optional_encoding, true) orelse {
+            return globalThis.throwInvalidArgumentTypeValue("buffer", "string or an instance of Buffer, TypedArray, DataView, or ArrayBuffer", input);
+        };
+
+        if (is_last)
+            this.has_called_end = true;
+
+        const err_buffer_too_large = globalThis.ERR_BUFFER_TOO_LARGE("Cannot create a Buffer larger than {d} bytes", .{this.maxOutputLength});
+
+        {
+            this.stream.write(input_to_queue.slice(), &this.output, false) catch |err| return handleTransformSyncStreamError(err, globalThis, this.stream.err_msg, &this.closed);
+            if (this.output.items.len > this.maxOutputLength) {
+                err_buffer_too_large.throw();
+                return .zero;
+            }
+            if (this.output.items.len > 0) event_loop.runCallback(push_fn, globalThis, thisctx, &.{this.collectOutputValue()});
+
+            while (!this.stream.broke_on_first_iter) {
```

Comment:
Can you make this name more clear? It's unclear what `broke_on_first_iter` means